src/scripts/downloadUiAssets.js

```diff
@@ -183,6 +205,48 @@ function getVersion(url) {
   }))
 }
 
+function checkUrls(baseUrl, path) {
+  console.log("Getting profile.js file from: ", baseUrl);
+  return (new Promise((resolve, reject) => {
+    request.get(baseUrl + path + 'profile.js', requestOptions, (err, data, body) => {
+      if (err) {
+        reject("profile.js file not found.");
+        return;
+      }
+      if (data.statusCode !== 200) {
+        console.log('Wrong path. Can not get the profile.js');
+       resolve('retry');
+      }
+      let profile = body;
+      resolve((profile && profile.includes('window.genesys.wwe.env.GWS_SERVICE_URL')));
+    })
+  }));
+}
+
+function getAuthURL(url) {
+  return (new Promise((resolve, reject) => {
+    console.log("Getting auth url from: ", url)
+    const redirect = url.replace('wwe-','gws.api01-');
```

Comment:
Why you don't use information from profile.js ?
Based on such kind of lines:
`window.genesys.wwe.env.EXPRESSION_WWE_URL_CAPTURING_GROUPS='https:\/\/wwe-([A-Za-z0-9]+)\.prod([0-9]+)\.genesysengage\.com\/.*';
window.genesys.wwe.env.GWS_SERVICE_URL='https://gws.api01-$WWE_URL_GROUP_1$.prod$WWE_URL_GROUP_2$.genesysengage.com';

window.genesys.wwe.env.EXPRESSION_WWE_URL_CAPTURING_GROUPS_1='https:\/\/(.*)-wwe-(.*)-pvt\.genesysengage\.com\/.*';
window.genesys.wwe.env.GWS_SERVICE_URL_1='https://$WWE_URL_GROUP_1$-gapi-$WWE_URL_GROUP_2$-pvt.genesysengage.com';`

Reply:
Fixed algorithm